framework/include/base/MooseObject.h

```diff
@@ -82,6 +152,9 @@ class MooseObject :
   /// The name of this object, reference to value stored in InputParameters
   const std::string & _name;
 
+  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
+  const bool & _enabled;
```

Comment:
Currently, we can enable/disable objects using time periods via TransientInterface (http://www.mooseframework.org/docs/doxygen/moose/classTransientInterface__inherit__graph_org.svg). What I am proposing here just makes this same idea a bit more general, but is essentially the same concept.